### PR TITLE
Support manual injection via annotation

### PR DESCRIPTION
**Pull request by mandopaloooza**
## Description
Allow the manual injection of SDP client using annotations. 

To enable manual injection, user is required to add the following annotations to the pod:
```yaml
  annotations:
    k8s.appgate.com/sdp-injector.device-id: "ba2349d9-03b2-428f-9d10-3c51638b2123"
    k8s.appgate.com/sdp-injector.client-config: "manual-inject-config"
    k8s.appgate.com/sdp-injector.client-secrets: "manual-inject-secret"
```

I've tested the setup with the following manifest:
```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: manual-inject-secret
  namespace: sdp-demo
type: Opaque
data:
  service-username: <USERNAME>
  service-password: <PASSWORD>
  service-url: <PROFILE_URL>
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: manual-inject-config
  namespace: sdp-demo
data:
  appgate-loglevel: INFO
---
apiVersion: v1
kind: Pod
metadata:
  name: manual-injection
  namespace: sdp-demo
  annotations:
    k8s.appgate.com/sdp-injector.device-id: "ba2349d9-03b2-428f-9d10-3c51638b2123"
    k8s.appgate.com/sdp-injector.client-config: "manual-inject-config"
    k8s.appgate.com/sdp-injector.client-secrets: "manual-inject-secret"
spec:
  containers:
  - name: dnsutils
    image: registry.k8s.io/e2e-test-images/jessie-dnsutils:1.3
    command:
      - sleep
      - "infinity"
    imagePullPolicy: IfNotPresent
  restartPolicy: Always
```

## Checklist
- [x] Bump `.version` and `.appVersion` in [k8s/crd/Chart.yaml](../k8s/crd/Chart.yaml)
- [x] Bump `.version` and `.appVersion` in [k8s/chart/Chart.yaml](../k8s/chart/Chart.yaml)

### CODE REVIEW

File: sdp-injector/src/injector.rs [hunk 1]
```diff
@@ -428,37 +428,38 @@ impl ServiceEnvironment {
...
-            client_secret_controller_url_key: pwd_field.to_string(),
-            client_secret_pwd_key: pwd_field,
-            client_secret_user_key: user_field,
-            client_device_id: device_id
-                .map(|s| s.clone())
-                .unwrap_or(uuid::Uuid::new_v4().to_string()),
-            n_containers: "0".to_string(),
-            k8s_dns_service_ip: None,
-        }))
+            .ok_or_else(|| "Pod was not found in the request")
```
Comment (lepiolet) [inline sdp-injector/src/injector.rs:437]: if you add `?;` you can avoid teh nested map

File: sdp-injector/src/injector.rs [hunk 2]
```diff
@@ -428,37 +428,38 @@ impl ServiceEnvironment {
...
+            .map(|p| {
+                // Need all three annotation to build service environment from pod
+                if let (Some(config), Some(secret), Some(device_id)) = (
```
Comment (lepiolet) [inline sdp-injector/src/injector.rs:440]: so now device id is mandatory, good.

File: sdp-injector/src/injector.rs [hunk 3]
```diff
@@ -428,37 +428,38 @@ impl ServiceEnvironment {
...
+                    p.annotation(SDP_ANNOTATION_CLIENT_SECRETS),
+                    p.annotation(SDP_ANNOTATION_CLIENT_DEVICE_ID),
+                ) {
+                    let service_id = request.service_id().ok()?;
+                    Some(ServiceEnvironment {
+                        service_name: service_id,
+                        client_config: config.to_string(),
+                        client_secret_name: secret.to_string(),
+                        client_secret_controller_url_key: "service-url".to_string(),
+                        client_secret_user_key: "service-username".to_string(),
```
Comment (lepiolet) [inline sdp-injector/src/injector.rs:451]: this change in names does not allow to reuse the same secret for different pods/services in the same ns, are we ok with that?

File: sdp-injector/src/injector.rs [hunk 4]
```diff
@@ -389,7 +389,7 @@ impl ServiceEnvironment {
         E: IdentityStore<ServiceIdentity>,
         R: ObjectRequest<Pod> + MaybeService + Annotated,
     {
-        if let Some(e) = ServiceEnvironment::from_pod(request)? {
+        if let Some(e) = ServiceEnvironment::from_pod(request) {
```
Comment (lepiolet) [inline sdp-injector/src/injector.rs:392]: why the change of signature in the return type?

**Review by lepiolet — COMMENTED** [review]
Why did you change the type of the function?

File: sdp-injector/src/injector.rs (hunk 4 above)
Comment (mandopaloooza) [reply sdp-injector/src/injector.rs:392]: I thought wrapping the Option with Result looked weird. For a valid response, from_pod would have to return 
```
           Ok(Some(ServiceEnvironment {
                service_name: service_id,
                client_config: config.to_string(),
                client_secret_name: secret.to_string(),
                client_secret_controller_url_key: "service-url".to_string(),
                client_secret_user_key: "service-username".to_string(),
                client_secret_pwd_key: "service-password".to_string(),
                client_device_id: device_id.to_string(),
                n_containers: "0".to_string(),
                k8s_dns_service_ip: None,
            }))
```
